Update hooks

fernandatoledo opened the pull request:
Delay, easing, and isInteraction options were added to the useAnimate hook.

Also, the AnimateParallel and AnimateSequence are moved to separate files and don't use Animate.parallel and Animate.sequence any more.

The parallelism is handled by starting animations simultaneously

The sequence is handled by starting the following animations as part of the current animation's callback.

This new way allows single animations callbacks to be triggered while being part of a parallel or sequence animation, that used to be impossible. This also allows single animations to be reused as the startAnimating method calls the reset before starting. We can easily embed now parallel in sequence animations or the inverse.

fernandatoledo committed:
Update hooks

Now every Animate.timing option is added as a useAnimate parameter.
Also the AnimateParallel and AnimateSequence are moved to separate files and dont use Animate.parallel and Animate.sequence any more, the parallelism and sequence is handled by starting animations simultaneously or in callbacks.

ManuViola77 commented on src/useAnimateParallel.js:
```diff
@@ -0,0 +1,50 @@
+  };
+
+  useEffect(() => {
+    animate && startAnimatingParallel({ callback: () => {} });
+  }, [animate]);
```
Do you need to send this empty callback when is already default to that in `startAnimatingParallel` definition?

ManuViola77 reviewed (approved):
Excellent!  really nice 💜  👏

fernandatoledo committed:
Fix callback param for startAnimating

Also add example of ParallelSequenceAnimation in case users want to check how to use a sequence inside a parallel or viceversa

fernandatoledo committed:
Add .expo folder to gitignore

xlasserre reviewed (approved):
Awesome! 👏
